internal/pkg/apply/apply_test.go

```diff
@@ -59,3 +66,321 @@ func TestApply(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, apply.Result{updatedObjects}, r)
 }
+
+func InitializeKustomizationWithPresence() ([]string, func(), error) {
+	f1, err := ioutil.TempDir("/tmp", "TestApply")
+	if err != nil {
+		return nil, nil, err
+	}
+	err = ioutil.WriteFile(filepath.Join(f1, "kustomization.yaml"), []byte(`apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+configMapGenerator:
+- name: test-map
+
+inventory:
+  type: ConfigMap
+  configMap:
+    name: inventory
+    namespace: default
+
+resources:
+- apply-service.yaml
+
+namespace: default
+`), 0644)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	err = ioutil.WriteFile(filepath.Join(f1, "apply-service.yaml"), []byte(`
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+spec:
+  selector:
+    app: MyApp
+  ports:
+  - protocol: TCP
+    port: 80
+    targetPort: 9376
+`), 0644)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	f2, err := ioutil.TempDir("/tmp", "TestApply")
+	if err != nil {
+		return nil, nil, err
+	}
+	err = ioutil.WriteFile(filepath.Join(f2, "kustomization.yaml"), []byte(`apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+configMapGenerator:
+- name: test-map
+  literals:
+  - foo=bar
+
+inventory:
+  type: ConfigMap
+  configMap:
+    name: inventory
+    namespace: default
+
+resources:
+- not-apply-service.yaml
+
+namespace: default
+`), 0644)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	err = ioutil.WriteFile(filepath.Join(f2, "not-apply-service.yaml"), []byte(`
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+  annotations:
+    kubectl.kubernetes.io/presence: EnsureDoesNotExist
+spec:
+  selector:
+    app: MyApp
+  ports:
+  - protocol: TCP
+    port: 80
+    targetPort: 9376
+`), 0644)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return []string{f1, f2}, func() {
+		os.RemoveAll(f1)
+		os.RemoveAll(f2)
+	}, nil
+}
+
+/* TestApplyWithPresenceAnnotation take following steps
```

Comment:
`Verify that Apply does not delete resources but ignores deleted resource.  Deletion should be delegated.`

Reply:
added